src/python/dxpy/scripts/dx.py

```diff
@@ -427,42 +414,29 @@ def logout(args):
         if state["interactive"]:
             dxpy.AUTH_HELPER = None
         else:
-            write_env_var("DX_SECURITY_CONTEXT", None)
+            dxpy.config.write("DX_SECURITY_CONTEXT", None)
 
 def set_api(protocol, host, port, write):
-    os.environ['DX_APISERVER_PROTOCOL'] = protocol
-    os.environ['DX_APISERVER_HOST'] = host
-    os.environ['DX_APISERVER_PORT'] = port
+    dxpy.config.update(DX_APISERVER_PROTOCOL=protocol,
+                       DX_APISERVER_HOST=host,
+                       DX_APISERVER_PORT=port)
     if write:
-        write_env_var("DX_APISERVER_PROTOCOL", protocol)
-        write_env_var("DX_APISERVER_HOST", host)
-        write_env_var("DX_APISERVER_PORT", port)
-    dxpy.set_api_server_info(host=host, port=port, protocol=protocol)
+        dxpy.config.save()
 
 def set_project(project, write, name=None):
     if dxpy.JOB_ID is None:
-        os.environ['DX_PROJECT_CONTEXT_ID'] = project
-        if name is not None:
-            set_env_var("DX_PROJECT_CONTEXT_NAME", name)
-        if write:
-            write_env_var("DX_PROJECT_CONTEXT_ID", project)
-            if name is not None:
-                write_env_var("DX_PROJECT_CONTEXT_NAME", name)
-            else:
-                try:
-                    os.remove(os.path.expanduser('~/.dnanexus_config/DX_PROJECT_CONTEXT_NAME'))
```

Comment:
Thanks for this refactoring, this is much better. Possibly minor point, is it important to distinguish between `DX_PROJECT_CONTEXT_NAME` being None and ""? Before you would remove the corresponding file if the value was set to None, but this is no longer done (with good reason, I think).